### ISSUE TITLE
Expose additional minigame metadata in foreground host activity

### ISSUE THREAD

**Issue by rikkolovescats** [bombsquad-community/plugin-manager#81]
In built-in minigames in Bombsquad, there is additional metadata that gets exposed which may then be accessed through plugins using the foreground host activity object, such as in meteor shower:
https://github.com/efroemling/ballistica/blob/4248540d887340de25e47539fdf567525ae8918e/assets/src/ba_data/python/bastd/game/meteorshower.py#L37-L49

which can be accessed through:
```python
>>> _ba.get_foreground_host_activity().allow_mid_activity_joins
False
```

Let's also make sure to have these fields in all minigames under plugin manager, so any local plugins that make use of these fields don't break. Such as zombie_horde currently only has the `name` field:
https://github.com/bombsquad-community/plugin-manager/blob/85e7f68797c4032f1b7d6f299b12d9d16f5ca66f/plugins/minigames/zombie_horde.py#L826-L832

**Comment by Freaku17** [comment]
If you take a closer look...
![IMG_20230118_170350](https://user-images.githubusercontent.com/92618708/213161254-36ff6850-8c71-4c88-bb11-1f19719c6f29.jpg)
![IMG_20230118_170415](https://user-images.githubusercontent.com/92618708/213161283-427ca766-d437-4904-a863-aa39011f8196.png)


I don't think without this `ba.TeamGameActivity` we can register a minigame...
And it already contains these fields in them...

So I don't believe there's any need to change anything, soooo close issue??

**Comment by rikkolovescats** [comment]
you're right, but it still seems to be missing the `allow_mid_activity_joins` field here:
https://github.com/bombsquad-community/plugin-manager/blob/85e7f68797c4032f1b7d6f299b12d9d16f5ca66f/plugins/minigames/zombie_horde.py#L153-L166

this minigame was only an example, there might be more minigames in plugin manager that i haven't checked yet that maybe missing a few fields, so missing fields would needed be added for those minigames as well

**Comment by Freaku17** [comment]
> you're right, but it still seems to be missing the `allow_mid_activity_joins` field here:
![IMG_20230118_221658](https://user-images.githubusercontent.com/92618708/213242947-6a56cfd3-b6d2-43ed-8fac-6cee8acd50d3.jpg)
![IMG_20230118_221716](https://user-images.githubusercontent.com/92618708/213242978-465ea2ab-03e2-4dfd-8456-a043eae243f1.jpg)

Which ultimately contains:
![IMG_20230118_221734](https://user-images.githubusercontent.com/92618708/213243086-e5cacae7-3938-434e-959f-8e1464f6594d.jpg)

**Comment by Freaku17** [comment]
> this minigame was only an example, there might be more minigames in plugin manager that i haven't checked yet that maybe missing a few fields, so missing fields would needed be added for those minigames as well

What I mean is a minigame can't be registered without this `ba.TeamGameActivity`, so ultimately all minigames does contain these information...

So no need for this "expose additional metadata"

**Comment by rikkolovescats** [comment]
yep, even still, the defaults may not be what we're looking for; zombie horde game doesn't have `allow_mid_activity_joins` defined, so the foreground activity will expose the default value which is `True` which could break any dependant plugins.

such as an AFK auto player remover plugin which doesn't want to remove players in activities that have `allow_mid_activity_joins` set to `False`, so that players aren't auto removed in elimination type games even if they're AFK, but such a plugin will still auto remove AFK players in zombie horde since zombie horde will default `allow_mid_activity_joins` to `True` which doesn't seem right as players have limited lives in zombie horde similar to elimination minigames.

**Comment by Freaku17** [comment]
> yep, even still, the defaults may not be what we're looking for; zombie horde game doesn't have `allow_mid_activity_joins` defined, so the foreground activity will expose the default value which is `True` which could break any dependant plugins.

Okay, and here it's developer's fault since ZombieHorde is like Elimination and shouldn't allow_mid_activity_joins 🤔

I understood, I think only in some games we'll have this issue 👍
